src/tests/functional/test_main.py

```diff
@@ -29,25 +31,234 @@
+def auth_token(pytestconfig):
+    """Get API token from command line"""
+    return pytestconfig.getoption("token")
+
+
+def _format_config(auth_token, feed_id):
+    """Configure for most of the test cases in this module."""
+    return (f"""id: cbc_binary_toolkit
+version: 0.0.1
+carbonblackcloud:
+  url: https://defense-eap01.conferdeploy.net
+  api_token: {auth_token}
+  org_key: WNEXFKQ7
+  ssl_verify: True
+  expiration_seconds: 3600
+database:
+  _provider: cbc_binary_toolkit.state.builtin.Persistor
+  location: ":memory:"
+engine:
+  name: Yara
+  feed_id: {feed_id}
+  type: local
+  _provider: cbc_binary_toolkit_examples.engine.yara_local.yara_engine.YaraFactory
+  rules_file: __file__/example_rule.yara
+    """)
+
+
+def _format_invalid_config(auth_token, feed_id):
+    """Configure for invalid test cases in this module."""
+    return (f"""id: cbc_binary_toolkit
+version: 0.0.1
+carbonblackcloud:
+  url: https://defense-eap01.conferdeploy.net
+  api_token: {auth_token}
+  org_key: WNEXFKQ7
+  ssl_verify: True
+  expiration_seconds: 3600
+database:
+  _provider: cbc_binary_toolkit.state.builtin.Persistor
+  location: ":memory:"
+engine:
+  name:
+  feed_id: {feed_id}
+  type: local
+  _provider: cbc_binary_toolkit_examples.engine.yara_local.yara_engine.YaraFactory
+  rules_file: __file__/example_rule.yara
+    """)
+
+
+def _create_feed(auth_token):
+    """Create a Feed to use for testing"""
+    # Feed Creation
+    url = "https://defense-eap01.conferdeploy.net/threathunter/feedmgr/v2/orgs/WNEXFKQ7/feeds"
+
+    payload = ("{\"feedinfo\": {\"name\": \"BATFuncTest\", \"owner\": \"DevRel\","
+               "\"provider_url\": \"some_url\", \"summary\": \"BAT functional testing\","
+               "\"category\": \"None\"},\n \"reports\": []}\n\n\n")
+    headers = {
+        'X-Auth-Token': f'{auth_token}',
+        'Content-type': 'application/json',
+        'Content-Type': 'text/plain'
+    }
+    response = requests.request("POST", url, headers=headers, data=payload)
+    return response.json()
+
+
+@pytest.fixture()
+def create_and_write_config(auth_token):
+    """Generate and write the test config to file"""
+    feed_info = _create_feed(auth_token)
+    feed_id = feed_info["id"]
+    with open("config/functional_config.yml", "w") as f:
+        config_text = _format_config(auth_token, feed_id)
+        f.write(config_text)
+    return feed_id
+
+
+@pytest.fixture()
+def create_and_write_invalid_config(auth_token):
+    """Generate and write the test config to file"""
+    feed_info = _create_feed(auth_token)
+    feed_id = feed_info["id"]
+    with open("config/functional_config.yml", "w") as f:
+        config_text = _format_invalid_config(auth_token, feed_id)
+        f.write(config_text)
+    return feed_id
+
+
+def get_reports_from_feed(auth_token, feed_id):
+    """GET to /feed/{feed_id}/reports to verify reports were sent"""
+    url = f'https://defense-eap01.conferdeploy.net/threathunter/feedmgr/v1/feed/{feed_id}/report'
```

Comment:
Any thoughts on using cbapi for creating, fetching and deleting feeds/reports? This would reduce the areas that we need to update when a service updates/deprecates?

Reply:
We include instructions for creating feed via Postman, CBAPI, and cURL in the Wiki. I'm not opposed to using CBAPI, and I agree that it'd be less work to maintain.

I chose the more manual method of creating/fetching/deleting a feed instead of CBAPI because I expected a real-world user to create their feed via Postman or something similar more than I expected them to use CBAPI. (I used the "generate code snippets" feature of Postman to create these functions)

Reply:
I could see the advanced users looking at our tests though the newer/average use would only use the wiki. My other thought is consistency with the core code and unit/integration tests